cmake/app/boilerplate.cmake

```diff
@@ -89,6 +89,8 @@ if((NOT DEFINED ZEPHYR_BASE) AND (DEFINED ENV_ZEPHYR_BASE))
   set(ZEPHYR_BASE ${ENV_ZEPHYR_BASE} CACHE PATH "Zephyr base")
 endif()
 
+find_package(ZephyrBuildConfiguration NAMES ZephyrBuild PATHS ${ZEPHYR_BASE}/../* QUIET NO_DEFAULT_PATH NO_POLICY_SCOPE)
```

Comment:
what happens if you have multiple apps with a ZephyrBuildConfiguration? Which one is picked? I assume this looks for the configuration under the workspace, right?

Reply:
ok, I think i get it now, the scope of this configuration is for the entire workspace, not one specific app, so probably there should be only one per workspace, right?

Reply:
right, and if you have multiple `ZephyrBuildConfig.cmake`, CMake will choose, based on order description here:
https://cmake.org/cmake/help/v3.17/variable/CMAKE_FIND_PACKAGE_SORT_ORDER.html
https://cmake.org/cmake/help/v3.17/variable/CMAKE_FIND_PACKAGE_SORT_DIRECTION.html

as this feature is intended as a hook for downstream users, they must also decide in their code if multiple `ZephyrBuildConfig.cmake` should be supported, and how.
I made a minimal example here https://github.com/zephyrproject-rtos/zephyr/pull/25575#issuecomment-634235501, that can be used as template and extended to simply return if the file should not be included, based on downstream criteria.

I could extend the docs with more advanced user examples at later time.
I think that could indeed be useful.